pkg/skaffold/deploy/deploy.go

```diff
@@ -22,13 +22,16 @@ import (
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/log"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/preview"
 )
 
 // Deployer is the Deploy API of skaffold and responsible for deploying
 // the build results to a Kubernetes cluster
 type Deployer interface {
 	log.Logger
 
+	preview.ResourcePreviewer
```

Comment:
love the package name `preview`. 
Should we rename it to `preview.Resource` to avoid stuttering

Reply:
we could, but the precedent is already set with `log.Logger`, `deploy.Deployer`, etc. I'm thinking we should stick to the convention we already have, but I'd be open to changing this later.

Reply:
I could be convinced to go with `preview.Previewer` since that's technically following our convention, but something about `Previewer` is kind of vague